Projects/Server/Utilities/ActivatorUtil.cs

```diff
@@ -42,32 +54,39 @@ public static ConstructorInfo GetConstructor(Type type, params Type[] args)
             // If there are more parameters, check if they any are not optional, if any are not, skip.
             // Otherwise all checks have passed. We have found a match 
             return args.Length <= paramList.Count || paramList.GetRange(args.Length, paramList.Count - args.Length).All(x => x.IsOptional);
-          }) ?? throw new Exception($"There is no empty/default constructor for {type}");
+          }) ?? throw new Exception($"There is no empty/default constructor for {type} that matches predicate.");
       }
       catch (Exception e)
       {
         throw new TypeInitializationException(type.ToString(), e);
       }
     }
-    public static object CreateInstance(Type type)
+
+    private static ConstructorInfo CheckPredicate(ConstructorInfo ci, Predicate<ConstructorInfo> predicate)
+    {

```

Comment:
This function seems unnecessary to me. When ever you want to check the predicate just do the following `predicate?.Invoke(ci) == true` or for a fail check: `predicate?.Invoke(ci) != true`
The only place I see the returned object used is on line 34. But that should probably be broken down into a proper if-else statement for clarity.